### PR TITLE
fix(Login): Google Login redirect to previously requested url

### LINKED ISSUE #175: Google login does not redirect to previously requested url
### Description

1. Log out of WISE
2. Go to the Authoring Tool url (or any other url in WISE that requires login) https://wise.berkeley.edu/teacher/edit/home
3. When it asks you to log in, choose "Sign in with Google"
4. Choose your Google account that has been used to register a WISE teacher account
5. You will be signed in and brought to the teacher home page but you should have been brought to the Authoring Tool

### PR DESCRIPTION
**Pull request by hirokiterashima**
## Changes
- Google login redirects to previously requested url
- Code clean up, remove unused code

## Test Prep
- Check out Client PR https://github.com/WISE-Community/WISE-Client/pull/798

## Test
- These should work as before:
   - Create new student and teacher account with Google, and log in with new account
   - Create new student and teacher account without Google, and log in with new account
- Teacher: repeat using non-Google account and Google account
   - log in from homepage => should take you to teacher home page
   - try to access /teacher/edit/home => should ask you to log in, and take you to authoring tool
- Student: repeat using non-Google account and Google account
   - log in from homepage => should take you to student home page
   - try to access /student/profile/edit => should ask you to log in, and take you to profile page

Closes #175

### CODE REVIEW

File: src/main/java/org/wise/portal/presentation/web/filters/WISEAuthenticationSuccessHandler.java [hunk 1]
```diff
@@ -147,33 +122,61 @@ public void onAuthenticationSuccess(HttpServletRequest request, HttpServletRespo
...
+    String redirectUrl = (String) request.getAttribute("redirectUrl");
+    if (redirectUrl != null) {
+      handleRedirectRequest(redirectUrl, request, response, userDetails, locale);
+    } else if (request.getServletPath().contains("google-login")) {
+      handleGoogleLogin(request, response, userDetails, locale);
+    }
+    if (ControllerUtil.isUserPreviousAdministrator()) {
+      response.sendRedirect(getUserHomeUrl(userDetails, locale));
+    }
+    //super.handle(request, response, authentication);
```
Comment (geoffreykwan) [inline src/main/java/org/wise/portal/presentation/web/filters/WISEAuthenticationSuccessHandler.java:135]: This commented out line of code can be deleted.

**Review by geoffreykwan — APPROVED** [review]
Looks good. I have a minor code clean up suggestion.

**Review by breity — APPROVED** [review]
Looks good.

**Pull request merged**